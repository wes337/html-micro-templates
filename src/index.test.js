var hMicroT = require('./index')

describe('stringToElement', () => {
    it('should replace delimited string with a button element', () => {
        const stringToConvert = 'There should be a ${buttonElement} here.'
        expect(hMicroT.stringToElement(
            { 
                string: stringToConvert,
                value: 'Button Text',
            }, 'button'
        )).toMatchSnapshot()
    })
    it('should replace delimited string with a self closing img element', () => {
        const stringToConvert = 'There should be a ${img} here.'
        expect(hMicroT.stringToElement(
            { 
                string: stringToConvert,
                value: 'somePic.jpg',
            }, 'img'
        )).toMatchSnapshot()
    })
})

describe('elementToString', () => {
    it('should replace element with a button delimited string', () => {
        const htmlToConvert = `"There should be a <button class='variable buttonElement'>Button Text</button> here, and a <section class='variable sectionElement'>section</section> here."`
        expect(hMicroT.elementToString(htmlToConvert)).toMatchSnapshot()
    })
})

describe('parseHTML', () => {
    const htmlToConvert = `"<span class='variable spanElement'>There should be a <button class='variable buttonElement'>Button Text</button> here.</span>"`
    it('should return array of literals', () => {
        expect(hMicroT.parseHTML(htmlToConvert)).toMatchSnapshot()
    })

    it('should return array of literals (with delimiters)', () => {
        expect(hMicroT.parseHTML(htmlToConvert, true)).toMatchSnapshot()
    })
})