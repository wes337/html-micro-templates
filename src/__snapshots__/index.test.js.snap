// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elementToString should replace element with a button delimited string 1`] = `"\\"There should be a \${buttonElement} here, and a \${sectionElement} here.\\""`;

exports[`parseHTML should return array of literals (with delimiters) 1`] = `
Array [
  "\${spanElement}",
  "\${buttonElement}",
]
`;

exports[`parseHTML should return array of literals 1`] = `
Array [
  "spanElement",
  "buttonElement",
]
`;

exports[`stringToElement should replace delimited string with a button element 1`] = `"There should be a <button class='variable buttonElement'>Button Text</button> here."`;

exports[`stringToElement should replace delimited string with a self closing img element 1`] = `"There should be a <img class='variable img' src='somePic.jpg'/> here."`;
